python/sqlflow_submitter/tensorflow/predict.py

```diff
@@ -71,22 +71,17 @@ def eval_input_fn(batch_size, cache=False):
     # NOTE: always use batch_size=1 when predicting to get the pairs of features and predict results
     #       to insert into result table.
     pred_dataset = eval_input_fn(1)
-    one_batch = pred_dataset.__iter__().next()
+    one_batch = next(pred_dataset)
     # NOTE: must run predict one batch to initialize parameters
     # see: https://www.tensorflow.org/alpha/guide/keras/saving_and_serializing#saving_subclassed_models
     classifier.predict_on_batch(one_batch[0])
     classifier.load_weights(save)
-    del pred_dataset
     pred_dataset = eval_input_fn(1, cache=True).make_one_shot_iterator()
     buff_rows = []
     column_names = feature_column_names[:]
     column_names.append(label_meta["feature_name"])
     with buffered_db_writer(conn.driver, conn, result_table, column_names, 100, hdfs_namenode_addr, hive_location, hdfs_user, hdfs_pass) as w:
-        while True:
-            try:
-                features = pred_dataset.get_next()
```

Comment:
@typhoonzero  I am curious why the original Python code is written in this way.

Reply:
Maybe I just kept the testing code when I trying to get the feature sample as well as predict results, anyway this is a cool fix.